pallets/node-manager/src/lib.rs

```diff
@@ -686,6 +698,74 @@ pub mod pallet {
 
             Ok(())
         }
+
+        #[pallet::call_index(5)]
+        #[pallet::weight(<T as Config>::WeightInfo::deregister_nodes(*number_of_nodes_to_deregister))]
```

Comment:
We could use `MaxNodesToDeregister` which is the maximum weight the call will cost, that the user sending the signed extrinsic will need to send, then allow, upon runtime execution, the proper weight/cost to be calculated and the appropriate gas fees paid.
It will also reduce the code checks.

Reply:
This way we calculate the cost accurately on the first go without charging more then refunding.